src/js/controller/preview/PreviewController.js

```diff
@@ -55,65 +66,100 @@
 
-    pskl.app.shortcutService.registerShortcut(this.onionSkinShortcut, this.toggleOnionSkin_.bind(this));
-    pskl.app.shortcutService.registerShortcut(this.originalSizeShortcut, previewSizeFunc.bind(this, PREVIEW.ORIGINAL));
-    pskl.app.shortcutService.registerShortcut(this.bestSizeShortcut, previewSizeFunc.bind(this, PREVIEW.BEST));
-    pskl.app.shortcutService.registerShortcut(this.fullSizeShortcut, previewSizeFunc.bind(this, PREVIEW.FULL));
+    for (var size in this.previewSize) {
+      if (this.previewSize.hasOwnProperty(size)) {
+        var wrapper = this.previewSize[size];
+        addEvent(wrapper.button, 'click', this.onChangePreviewSize_, this, wrapper.name);
+        registerShortcut(wrapper.shortcut, this.onChangePreviewSize_.bind(this, wrapper.name));
+        var tooltip = pskl.utils.TooltipFormatter.format(wrapper.tooltip, wrapper.shortcut);
+        wrapper.button.setAttribute('title', tooltip);
+      }
+    }
 
     $.subscribe(Events.FRAME_SIZE_CHANGED, this.onFrameSizeChange_.bind(this));
     $.subscribe(Events.USER_SETTINGS_CHANGED, $.proxy(this.onUserSettingsChange_, this));
     $.subscribe(Events.PISKEL_SAVE_STATE, this.setRenderFlag_.bind(this, true));
     $.subscribe(Events.PISKEL_RESET, this.setRenderFlag_.bind(this, true));
 
-    this.initTooltips_();
-    this.initDynamicPreviewSize_();
+    this.updatePreviewSizeButtons_();
     this.popupPreviewController.init();
 
     this.updateZoom_();
     this.updateOnionSkinPreview_();
-    this.updateOriginalSizeButton_();
+    this.selectPreviewSizeButton_();
     this.updateMaxFPS_();
     this.updateContainerDimensions_();
   };
 
-  ns.PreviewController.prototype.initTooltips_ = function () {
-    var onionSkinTooltip = pskl.utils.TooltipFormatter.format('Toggle onion skin', this.onionSkinShortcut);
-    this.toggleOnionSkinButton.setAttribute('title', onionSkinTooltip);
-    var originalSizeTooltip = pskl.utils.TooltipFormatter.format('Original size preview', this.originalSizeShortcut);
-    this.originalSizeButton.setAttribute('title', originalSizeTooltip);
-    var bestSizeTooltip = pskl.utils.TooltipFormatter.format('Round factor size preview', this.bestSizeShortcut);
-    this.bestSizeButton.setAttribute('title', bestSizeTooltip);
-    var fullSizeTooltip = pskl.utils.TooltipFormatter.format('Biggest factor size preview', this.fullSizeShortcut);
-    this.fullSizeButton.setAttribute('title', fullSizeTooltip);
+  ns.PreviewController.prototype.updatePreviewSizeButtons_ = function () {
+    var fullZoom = this.calculateZoom_();
+    var seamlessModeEnabled = pskl.UserSettings.get(pskl.UserSettings.SEAMLESS_MODE);
+
+    if (seamlessModeEnabled) {
+      this.togglePreviewSizeButtonState_(this.previewSize.best, false);
+      this.togglePreviewSizeButtonState_(this.previewSize.full, false);
+      this.onChangePreviewSize_(this.previewSize.original.name);
+    } else {
+      if (Number.isInteger(fullZoom)) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger 
Not available everywhere yet. Either drop the polyfill in utils/core.js or directly use `Math.floor(fullZoom) === fullZoom` here.

That's the reason why the tests are failing on Travis.